### PR TITLE
Fix a crash for when the root tree has no children

### PR DESCRIPTION
**Pull request by hughsie**
This will only happen when using xb_node_get_child(xb_silo_get_root()).

Fixes https://github.com/hughsie/libxmlb/issues/120

### CODE REVIEW

**Comment by hughsie** [comment]
@mcrha can you verify please.

**Comment by mcrha** [comment]
When you apply this [xb.patch.txt](https://github.com/hughsie/libxmlb/files/8390167/xb.patch.txt) on top of this merge request, then it'll reproduce the crash.

As the `xb_silo_get_root` returns a valid node, it should be able to use all the `XbNode` methods, right?

**Comment by mcrha** [comment]
And this patch [xb.patch.txt](https://github.com/hughsie/libxmlb/files/8390284/xb.patch.txt) extends the previous patch to reproduce the `strtab incorrect` error - that might need a new test, I guess.

File: src/xb-silo.c [hunk 1]
```diff
@@ -435,6 +435,8 @@ xb_silo_get_node_tail(XbSilo *self, XbSiloNode *n)
 const gchar *
 xb_silo_get_node_element(XbSilo *self, XbSiloNode *n)
 {
+	if (n == NULL)
+		return NULL;
```
Comment (mcrha) [inline src/xb-silo.c:439]: I believe it'll be a good idea to be proactive and to catch all the `xb_node_get_` functions in this merge request, not only those I happened to use.

**Comment by mcrha** [comment]
Still not perfect:
```
Thread 36 "gnome-software" received signal SIGSEGV, Segmentation fault.
[Switching to Thread 0x7fff51ca3640 (LWP 18027)]
0x00007ffff6e14ac4 in xb_silo_node_has_flag (self=0x0, flag=XB_SILO_NODE_FLAG_IS_ELEMENT) at ../src/xb-silo-node.c:39
39		return (self->flags & flag) > 0;
#0  0x00007ffff6e14ac4 in xb_silo_node_has_flag (self=0x0, flag=XB_SILO_NODE_FLAG_IS_ELEMENT) at ../src/xb-silo-node.c:39
#1  0x00007ffff6e14a6e in xb_silo_node_get_size (self=0x0) at ../src/xb-silo-node.c:17
#2  0x00007ffff6e0feca in xb_silo_get_child_node (self=0xd41df0, n=0x0) at ../src/xb-silo.c:277
#3  0x00007ffff6e0be66 in xb_node_child_iter_init (iter=0x7fff51ca24f0, self=0x7fff580e4040) at ../src/xb-node.c:306
```

**Comment by hughsie** [comment]
> Still not perfect

Aha, of course. One more commit to test please.

**Comment by mcrha** [comment]
```
Thread 35 "gnome-software" received signal SIGSEGV, Segmentation fault.
[Switching to Thread 0x7fff524a4640 (LWP 63838)]
0x00007ffff6e0fe8a in xb_silo_get_next_node (self=0xd42600, n=0x0) at ../src/xb-silo.c:266
266		if (n->next == 0x0)
#0  0x00007ffff6e0fe8a in xb_silo_get_next_node (self=0xd42600, n=0x0) at ../src/xb-silo.c:266
#1  0x00007ffff6e0bc75 in xb_node_get_next (self=0x7fff580e3510) at ../src/xb-node.c:227
```

It would really be beneficial to cover all the `priv->silo, priv->sn` usages, otherwise it's just that it'll work for calls my code does, but can be broken for someone else, using different calls.

**Comment by hughsie** [comment]
> It would really be beneficial to cover all the priv->silo, priv->sn usages

Okay, agreed. Done.

**Comment by mcrha** [comment]
This works fine here, no crash spotted so far. Thank you.

**Pull request merged**